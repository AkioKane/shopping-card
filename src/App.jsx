import { useState, useEffect } from 'react';
import { Outlet, Link } from 'react-router-dom';
import iconHead from "./assets/paw.png";
import home from "./assets/home.svg";
import homePink from "./assets/home-pink.svg";
import shop from "./assets/store.svg";
import shopPink from "./assets/store-pink.svg";
import cart from "./assets/shopping_cart.svg";
import cartPink from "./assets/shopping_cart_pink.svg";
import './styles/App.css';

async function getData(category) {
  const response = await fetch(
    `https://fakestoreapi.in/api/products/category?type=${category}`, 
    {mode: "cors"}
  );

  const data = await response.json();
  return data.products;
}

function App() {
  const [dataMobile, setDataMobile] = useState(null);
  const [dataAudio, setDataAudio] = useState(null);
  const [dataGaming, setDataGaming] = useState(null);
  const [dataLaptop, setDataLaptop] = useState(null);
  const [dataTv, setDataTv] = useState(null);
  const [dataAppliances, setDataAppliances] = useState(null);
  const [activeComponent, setActiveComponent] = useState("home");
  const [hoverElement, setHoverElement] = useState(null);
  const [cartList, setCartList] = useState([]);

  useEffect(() => {
    async function fetchData() {
      setDataMobile(await getData("mobile"));
      setDataAudio(await getData("audio"));
      setDataGaming(await getData("gaming"));
      setDataLaptop(await getData("laptop"));
      setDataTv(await getData("tv"));
      setDataAppliances(await getData("appliances"));
    }
    fetchData();
  }, []);

  return (
    <>
      <div className="main">
        <aside className="menu">
          <div className="head">
            <h1>
              <span
                style={{
                  color: "#ff798f",
                  margin: 0,
                  marginRight: "0.1em",
                }}
              >
                Paw
              </span>
              Store
            </h1>
            <img className="logo-head" src={iconHead} alt="icon" />
          </div>

          <Link
            to={"/"}
            onClick={() => {
              setActiveComponent("home");
            }}
            onMouseOver={() => {
              setHoverElement("home");
            }}
            onMouseOut={() => {
              setHoverElement(null);
            }}
            style={{
              color: activeComponent === "home" ? "#ff798f" : "white",
              transform: activeComponent === "home" ? "scale(1.1)" : "none",
              transition: "all 0.1s ease-in-out",
            }}
          >
            <h2>
              <img
                className="icon"
                src={
                  activeComponent === "home" || hoverElement === "home"
                    ? homePink
                    : home
                }
                alt="home"
              />
              Home
            </h2>
          </Link>

          <Link
            to={"shop"}
            onClick={() => {
              setActiveComponent("shop");
            }}
            onMouseOver={() => {
              setHoverElement("shop");
            }}
            onMouseOut={() => {
              setHoverElement(null);
            }}
            style={{
              color: activeComponent === "shop" ? "#ff798f" : "white",
              transform: activeComponent === "shop" ? "scale(1.1)" : "none",
              transition: "all 0.1s ease-in-out",
            }}
          >
            <h2>
              <img
                className="icon"
                src={
                  activeComponent === "shop" || hoverElement === "shop"
                    ? shopPink
                    : shop
                }
                alt="cart"
              />
              Shop
            </h2>
          </Link>
          <Link
            to={"cart"}
            onClick={() => {
              setActiveComponent("cart");
            }}
            onMouseOver={() => {
              setHoverElement("cart");
            }}
            onMouseOut={() => {
              setHoverElement(null);
            }}
            style={{
              color: activeComponent === "cart" ? "#ff798f" : "white",
              transform: activeComponent === "cart" ? "scale(1.1)" : "none",
              transition: "all 0.1s ease-in-out",
            }}
          >
            <h2>
              <img
                className="icon"
                src={
                  activeComponent === "cart" || hoverElement === "cart"
                    ? cartPink
                    : cart
                }
                alt="cart"
              />
              Cart
            </h2>
          </Link>
        </aside>

        <div className="content">
          <Outlet
            context={{
              setActiveComponent,
              dataMobile,
              dataAudio,
              dataGaming,
              dataLaptop,
              dataTv,
              dataAppliances,
              cartList,
              setCartList,
            }}
          />
        </div>
      </div>
    </>
  );
}

export default App
