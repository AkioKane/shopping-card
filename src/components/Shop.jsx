import { useEffect } from "react";
import { useOutletContext } from "react-router-dom";
import Card from "./Card";
import "../styles/Shop.css";

function Shop() {
  const {
    setActiveComponent,
    dataMobile,
    dataAudio,
    dataGaming,
    dataLaptop,
    dataTv,
    dataAppliances,
    setCartList,
    cartList,
  } = useOutletContext();

  useEffect(() => {
    setActiveComponent("shop")
  }, [])

  return (
    <>
      <div className="shop-content">
        <div className="container-shop sneakers">
          <h1 className="header-shop">Mobiles</h1>
          <div className="cards">
            {dataMobile ? [1, 2, 3, 4, 5, 6, 7, 8, 9, 10].map((id) => (
              <Card key={id} 
                dataCategory={dataMobile ? dataMobile : null} 
                id={id} 
                cartList={cartList}
                setCartList={setCartList}
              />
            )) : ""}
          </div>
        </div>
        <div className="container-shop audio">
          <h1 className="header-shop">Audio</h1>
          <div className="cards">
            {dataAudio ? [1, 2, 3, 4, 5, 6, 7, 8, 9, 10].map((id) => (
              <Card key={id} 
                dataCategory={dataAudio ? dataAudio : null} 
                id={id} 
                cartList={cartList}
                setCartList={setCartList}
              />
            )) : ""}
          </div>
        </div>
        <div className="container-shop gaming">
          <h1 className="header-shop">Gaming</h1>
          <div className="cards">
            {dataGaming ? [1, 2, 3, 4, 5, 6, 7, 8, 9, 10].map((id) => (
              <Card key={id} 
                dataCategory={dataGaming ? dataGaming : null} 
                id={id} 
                cartList={cartList}
                setCartList={setCartList}
              />
            )) : ""}
          </div>
        </div>
        <div className="container-shop laptop">
          <h1 className="header-shop">Laptop</h1>
          <div className="cards">
            {dataLaptop ? [1, 2, 3, 4, 5, 6, 7, 8, 9, 10].map((id) => (
              <Card key={id} 
                dataCategory={dataLaptop ? dataLaptop : null} 
                id={id} 
                cartList={cartList}
                setCartList={setCartList}
              />
            )) : ""}
          </div>
        </div>
        <div className="container-shop tv">
          <h1 className="header-shop">TV</h1>
          <div className="cards">
            {dataTv ? [1, 2, 3, 4, 5, 6, 7, 8, 9, 10].map((id) => (
              <Card key={id} 
                dataCategory={dataTv ? dataTv : null} 
                id={id} 
                cartList={cartList}
                setCartList={setCartList}
              />
            )) : ""}
          </div>
        </div>
        <div className="container-shop appliances">
          <h1 className="header-shop">Appliances</h1>
          <div className="cards">
            {dataAppliances ? [1, 2, 3, 4, 5, 6, 7, 8, 9, 10].map((id) => (
              <Card key={id} 
                dataCategory={dataAppliances ? dataAppliances : null} 
                id={id} 
                cartList={cartList}
                setCartList={setCartList}
              />
            )) : ""}
          </div>
        </div>
      </div>
    </>
  );
}

export default Shop;