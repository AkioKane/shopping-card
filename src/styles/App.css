.main {
  display: flex;
  align-items: center;
  min-height: 100vh;
}

.menu {
  position: fixed;
  top: 0;
  left: 0;
  display: flex;
  flex-direction: column;
  width: var(--size-sidebar-width);
  height: 100vh;
  background: linear-gradient(45deg, rgba(28,25,23,1) 0%, rgba(96,0,16,1) 52%, rgba(47,0,8,1) 97%);
  color: white;
  font-family: "Ubuntu";
  box-shadow: 0 0 4px #383838;
  transition: all 1s ease-in-out;
  z-index: 2;
  
  * {
    margin-left: 2em;
  }

  h1,
  a,
  .logo-head {
    margin: 0;
  }

  .head {
    margin: 0;
  }
}

.black-background {
  position: fixed;
  top: 0;
  width: 100vw;
  height: 100vh;
  background-color: black;
  transition: all 1s linear;
  z-index: 1;
}

.menu .head {
  display: flex;
  align-items: center;
  justify-content: center;
  width: 100%;
  font-family: "Park";
  font-weight: 300;
  margin-top: 1em;
  margin-bottom: 2em;
  gap: 0.8em;
  font-size: 20px;
}

.menu a {
  text-decoration: none;
  color: white;
}

.menu hr {
  width: 100%;
  margin-top: 0.25em;
  margin-left: 0;
  margin-bottom: 1.5em;
  margin-right: 0;
}

.menu h2 {
  display: flex;
  align-items: center;
  margin-bottom: 0.25em;
}

.menu .head img {
  display: flex;
  justify-content: center;
  align-items: center;
  padding: 5px;
  background-color: white;
  border-radius: 50%;
  height: 42px;
  transform: rotate(0.5deg);
  animation: spin 13s linear infinite;
}

.menu .head h1 {
  transition: all 0.3s ease-in-out;
}

.icon {
  margin: 0;
  margin-right: 5px;
}

.content {
  display: flex;
  flex-direction: column;
  align-items: center;
  z-index: 0;
}

.logo-head {
  transition: all 0.1s ease-in-out;
}

.logo-head:hover {
  scale: 1.2;
}

.show-sidebar img {
  margin-left: 0.25em;
  margin-right: 0.25em;
  margin-top: 0.15em;
  margin-bottom: 0.15em;
}

.show-sidebar {
  position: fixed;
  top: 2em;
  transition: all 1s ease-in-out;
  background-color: #383838;
  border: none;
  padding: 5px;
  border-radius: 0 16px 16px 0;
  z-index: 1;
}

@keyframes spin {
  0% {
    transform: rotate(0deg);
  }

  100% {
    transform: rotate(360deg);
  }
}