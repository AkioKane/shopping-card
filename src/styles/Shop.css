.shop-content {
  display: flex;
  flex-direction: column;
  align-items: center;
  margin-bottom: 2em;
  width: var(--content-head-width);
}

.container-shop {
  display: flex;
  flex-direction: column;
  align-items: center;
  width: 100%;
  margin-top: 2em;
}

.shop-content .header-shop {
  display: flex;
  justify-content: center;
  align-items: center;
  background: rgb(36, 32, 30);
  color: white;
  font-size: 52px;
  margin-bottom: 1.5em;
  margin-top: 1em;
  border-radius: 16px;
}