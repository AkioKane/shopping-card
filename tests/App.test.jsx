import { describe, it, expect } from 'vitest';
import { getByText, getByRole, render, screen, waitFor } from "@testing-library/react";
import { BrowserRouter, RouterProvider, createMemoryRouter } from 'react-router-dom';
import userEvent from '@testing-library/user-event';
import App from "../src/App";
import Shop from "../src/components/Shop";

const renderWithRouter = (ui) => {
  return render(<BrowserRouter>{ui}</BrowserRouter>);
};


describe('App component', () => {
  it("render Sidebar", () => {
    renderWithRouter(<App />);
    
    expect(
      screen.getByText(
        /Home/i
      )
    ).toBeInTheDocument();
  });

  it("click view btn", async () => {
    const user = userEvent.setup();

    const { container } = renderWithRouter(<App />);

    const button = screen.getByText('Shop').closest('a')

    await user.click(button);

    expect(document.querySelector("head")).toBeInTheDocument();
  });
});