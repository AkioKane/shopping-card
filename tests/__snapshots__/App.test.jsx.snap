// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`App component > render 1`] = `
<div>
  <div
    class="main"
  >
    <aside
      class="menu"
      id="main"
      style="left: 0px;"
    >
      <div
        class="head"
      >
        <h1>
          <span
            style="color: rgb(255, 121, 143); margin: 0px 0.1em 0px 0px;"
          >
            Paw
          </span>
          Store
        </h1>
        <img
          alt="icon"
          class="logo-head"
          src="/src/assets/paw.png"
        />
      </div>
      <a
        data-discover="true"
        href="/"
        style="color: rgb(255, 121, 143); transform: scale(1.1); transition: all 0.1s ease-in-out;"
      >
        <h2>
          <img
            alt="home"
            class="icon"
            src="/src/assets/home-pink.svg"
          />
          Home
        </h2>
      </a>
      <a
        data-discover="true"
        href="/shop"
        style="color: white; transform: none; transition: all 0.1s ease-in-out;"
      >
        <h2>
          <img
            alt="cart"
            class="icon"
            src="/src/assets/store.svg"
          />
          Shop
        </h2>
      </a>
      <a
        data-discover="true"
        href="/cart"
        style="color: white; transform: none; transition: all 0.1s ease-in-out;"
      >
        <h2>
          <img
            alt="cart"
            class="icon"
            src="/src/assets/shopping_cart.svg"
          />
          Cart
        </h2>
      </a>
    </aside>
    <div
      class="black-background"
      style="opacity: 0; z-index: -1;"
    />
    <div
      class="content"
      style="margin-left: calc(300px + 5em);"
    />
  </div>
</div>
`;

exports[`App component > render home page 1`] = `
<div>
  <div
    class="main"
  >
    <aside
      class="menu"
      id="main"
      style="left: 0px;"
    >
      <div
        class="head"
      >
        <h1>
          <span
            style="color: rgb(255, 121, 143); margin: 0px 0.1em 0px 0px;"
          >
            Paw
          </span>
          Store
        </h1>
        <img
          alt="icon"
          class="logo-head"
          src="/src/assets/paw.png"
        />
      </div>
      <a
        data-discover="true"
        href="/"
        style="color: rgb(255, 121, 143); transform: scale(1.1); transition: all 0.1s ease-in-out;"
      >
        <h2>
          <img
            alt="home"
            class="icon"
            src="/src/assets/home-pink.svg"
          />
          Home
        </h2>
      </a>
      <a
        data-discover="true"
        href="/shop"
        style="color: white; transform: none; transition: all 0.1s ease-in-out;"
      >
        <h2>
          <img
            alt="cart"
            class="icon"
            src="/src/assets/store.svg"
          />
          Shop
        </h2>
      </a>
      <a
        data-discover="true"
        href="/cart"
        style="color: white; transform: none; transition: all 0.1s ease-in-out;"
      >
        <h2>
          <img
            alt="cart"
            class="icon"
            src="/src/assets/shopping_cart.svg"
          />
          Cart
        </h2>
      </a>
    </aside>
    <div
      class="black-background"
      style="opacity: 0; z-index: -1;"
    />
    <div
      class="content"
      style="margin-left: calc(300px + 5em);"
    />
  </div>
</div>
`;
